src/groups/GroupList.test.tsx

```diff
@@ -0,0 +1,186 @@
+        ({
+          Groups: {
+            list: jest.fn().mockResolvedValue([memberGroup, adminGroup]),
+          },
+        } as DeepPartial<AjaxContract> as AjaxContract)
+    );
+    asMockedFn(getStateHistory).mockReturnValue({});
+    const { getByText, queryByText, getByLabelText } = render(<GroupList />);
+    await waitFor(() => expect(getByText(memberGroup.groupName, { exact: false })).toBeDefined());
+    await waitFor(() => expect(getByText(adminGroup.groupName, { exact: false })).toBeDefined());
+    // Act
+    fireEvent.change(getByLabelText('Search groups'), { target: { value: memberGroup.groupName } });
+
+    // Assert
+    await waitFor(() => expect(getByText(memberGroup.groupName, { exact: false })).toBeDefined());
+    await waitFor(() => expect(queryByText(adminGroup.groupName, { exact: false })).toBeFalsy());
+    await waitFor(() => expect(asMockedFn(updateStateHistory)).toHaveBeenCalledWith({ filter: memberGroup.groupName }));
+  });
+
+  it('sets the correct group to delete', async () => {
+    // Arrange
+    asMockedFn(Ajax).mockImplementation(
+      () =>
+        ({
+          Groups: {
+            list: jest.fn().mockResolvedValue([memberGroup, adminGroup]),
+          },
+        } as DeepPartial<AjaxContract> as AjaxContract)
+    );
+    const user = userEvent.setup();
+    const { getByText, queryByText, getByRole } = render(<GroupList />);
+    await waitFor(() => expect(getByText(adminGroup.groupName, { exact: false })).toBeDefined());
+    const menuLabel = `Action Menu for Group: ${adminGroup.groupName}`;
+    const menu = getByRole('button', { name: menuLabel });
+    expect(menu).toBeDefined();
+
+    // Act
+    await user.click(menu);
+    const deleteButton = getByText('Delete', { exact: false });
+    expect(deleteButton).toBeDefined();
+    await user.click(deleteButton);
+    // Assert
+    // waiting for the modal to appear
+    waitFor(() => expect(queryByText('Delete group')).toBeDefined());
+  });
+
+  it('sets the correct group to leave', async () => {
+    // Arrange
+    asMockedFn(Ajax).mockImplementation(
+      () =>
+        ({
+          Groups: {
+            list: jest.fn().mockResolvedValue([memberGroup, adminGroup]),
+          },
+        } as DeepPartial<AjaxContract> as AjaxContract)
+    );
+    const user = userEvent.setup();
+    const { getByText, queryByText, getByRole } = render(<GroupList />);
+    await waitFor(() => expect(getByText(adminGroup.groupName, { exact: false })).toBeDefined());
+    const menuLabel = `Action Menu for Group: ${adminGroup.groupName}`;
+    const menu = getByRole('button', { name: menuLabel });
+    expect(menu).toBeDefined();
+
+    // Act
+    await user.click(menu);
+    const leaveButton = getByText('Leave', { exact: false });
+    expect(leaveButton).toBeDefined();
+    await user.click(leaveButton);
+
+    // Assert
+    // waiting for the modal to appear
+    waitFor(() => expect(queryByText('Leave group')).toBeDefined());
```

Comment:
The test case says "sets the correct group to leave". Did you mean to assert something about which "Leave group" option is being shown?

Reply:
Should be a bit more meaningful, now.